### ISSUE TITLE
New crash in subarray test on 64-bit

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#15508]
One commit in the range fc469b68316061d95003cb0768c43dbc4b1efd0f...64dbecf6ca76a46be34c3addd39cd516f383b70a (i.e. https://github.com/JuliaLang/julia/pull/15444 by @vtjnash  -- again, sorry! :-) introduced a crash in the subarray tests when building the RPM nightlies on 64-bit (but not 32-bit). This is with system LLVM 3.8 plus the Julia patch.

```
signal (11): Segmentation fault
while loading /builddir/build/BUILD/julia/test/subarray.jl, in expression starting on line 309
[inline] at ./essentials.jl:25
reindex at ./subarray.jl:130
unknown function (ip: 0x7f03698fb260)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
[inline] at ./subarray.jl:91
sub at ./subarray.jl:193
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
jl_f__apply at /builddir/build/BUILD/julia/src/builtins.c:512
runtests at /builddir/build/BUILD/julia/test/subarray.jl:221
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
[inline] at ./int.jl:28
runviews at /builddir/build/BUILD/julia/test/subarray.jl:249
unknown function (ip: 0x7f0369b236b3)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
[inline] at /builddir/build/BUILD/julia/test/subarray.jl:329
anonymous at ./<no file>:4294967295
unknown function (ip: 0x7f0373dcb420)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/toplevel.c:552
jl_parse_eval_all at /builddir/build/BUILD/julia/src/ast.c:784
[inline] at ./essentials.jl:79
include_string at ./loading.jl:371
unknown function (ip: 0x7f03b06a3194)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
include_from_node1 at ./loading.jl:420
unknown function (ip: 0x7f05b2593b75)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
[inline] at ./util.jl:180
runtests at /builddir/build/BUILD/julia/test/testdefs.jl:7
#16 at /builddir/build/BUILD/julia/test/runtests.jl:36
unknown function (ip: 0x7f03b0681c33)
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
jl_f__apply at /builddir/build/BUILD/julia/src/builtins.c:512
run_work_thunk at ./multi.jl:744
[inline] at ./multi.jl:1040
#255 at ./task.jl:59
[inline] at /builddir/build/BUILD/julia/src/julia_internal.h:69
jl_call_method_internal at /builddir/build/BUILD/julia/src/gf.c:1848
start_task at /builddir/build/BUILD/julia/src/task.c:245
unknown function (ip: (nil))
Allocations: 844523643 (Pool: 844164628; Big: 359015); GC: 2060
    From worker 3:       * subarray             Worker 3 terminated.
```

https://copr-be.cloud.fedoraproject.org/results/nalimilan/julia-nightlies/fedora-23-x86_64/00168226-julia/build.log.gz

**Comment by tkelman** [comment]
Which Julia patch for 3.8? With `reindex` showing up in the backtrace this looks similar to what I was seeing on win64 in https://github.com/JuliaLang/julia/pull/15417

**Comment by nalimilan** [comment]
I only had D17165, didn't notice the new one. I'm currently retrying with the threads patch.

**Comment by tkelman** [comment]
The patch that turns of thread-dependent code in LLVM shouldn't really be necessary anywhere other than Windows (if it is disabling code that we otherwise could use elsewhere, I can make it more selectively applied), just saying that the cause might be similar and possibly unrelated to any patches we're carrying.

**Comment by nalimilan** [comment]
OK, I canceled the build then. I didn't mean that the patch was in cause, I rather wanted to exclude the possibility that the crash would be due to its absence.

Note that it succeeds with LLVM 3.7.1.

**Comment by KristofferC** [comment]
Is this still relevant?

**Comment by nalimilan** [comment]
Looks like it's been fixed either in Julia or in LLVM 3.8.1.

**Comment by tkelman** [comment]
probably fixed by https://github.com/JuliaLang/julia/pull/16503
